internal/observability/middleware.go

```diff
@@ -181,8 +181,12 @@ func applyLogLevelsConfig(dst *levels, src *DirectionalLevelsConfig) {
 
 // Handle implements middleware.UnaryInbound.
 func (m *Middleware) Handle(ctx context.Context, req *transport.Request, w transport.ResponseWriter, h transport.UnaryHandler) error {
-	call := m.graph.begin(ctx, transport.Unary, _directionInbound, req)
-	defer m.handlePanicForCall(call, transport.Unary)
+
+	var call call
+	m.graph.begin(ctx, transport.Unary, _directionInbound, req, &call)
+	// It is safe to pass a pointer to defer because the stack memory doesn't get released so there is no escape to the
+	// heap
```

Comment:
I don't understand - is there a guarantee that the call won't get heap allocated anyway?

Separately, what is the concern? That this memory is on a heap, so you're passing a pointer to memory that might get released (?)

Reply:
is there a guarantee that the call won't get heap allocated anyway?
- No, there is no guarantee if someone changes the code, but as of now it would always be on the stack because its liveness can be determined at compile time.

Separately, what is the concern? That this memory is on a heap, so you're passing a pointer to memory that might get released (?)
- Concern? Performance impact. Heap allocation is more expensive than stack, but other than that there is no bug/issue if it is either on the stack or the heap.

Reply:
So let's keep it on stack then and return call from graph like it was before ? 
Since zap fields are removed, copy cost is going to be low now and it's going to happen only once per call, so it should be okay. 

Can you separately benchmark this after making call & zap changes and if graph changes actually improve performance at all ?